tiledb/CMakeLists.txt

```diff
@@ -540,31 +540,13 @@ endif()
 if (TILEDB_AZURE)
   message(STATUS "The TileDB library is compiled with Azure support.")
 
-  if (TILEDB_VCPKG)
-    find_package(azure-storage-blobs-cpp CONFIG REQUIRED)
-    target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
-            INTERFACE
-            Azure::azure-storage-blobs
-            )
-  else()
-    find_package(AzureCore_EP REQUIRED)
-    find_package(AzureStorageCommon_EP REQUIRED)
-    find_package(AzureStorageBlobs_EP REQUIRED)
-    if (NOT WIN32)
-      if (NOT TARGET LibXml2::LibXml2)
-        find_package(LibXml2 REQUIRED)
-      endif()
-    endif()
-    target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
-            INTERFACE
-            Azure::azure-storage-blobs
-            Azure::azure-storage-common
-            Azure::azure-core)
-    # Link xml after Azure to allow symbol resolution on all platforms
-    if (NOT WIN32)
-      target_link_libraries(TILEDB_CORE_OBJECTS_ILIB INTERFACE LibXml2::LibXml2)
-    endif()
-  endif()
```

Comment:
This part is used only if vcpkg is disabled, [which no longer happens in first-party code](https://github.com/search?q=-DTILEDB_VCPKG%3DOFF+NOT+is%3Afork&type=code).

Regarding the new dependency, Conda will need to be updated to add `azure-identity-cpp` in the requirements, and an empty vcpkg port overlay.